servers/ramatak/ad_server.h

```diff
@@ -0,0 +1,81 @@
+#ifndef _AD_SERVER_H
+#define _AD_SERVER_H
+
+#include "core/class_db.h"
+#include "core/list.h"
+#include "core/math/random_pcg.h"
+#include "core/object.h"
+#include "core/reference.h"
+
+class AdPlugin;
+
+class AdServer : public Object {
+	GDCLASS(AdServer, Object);
+
+public:
+	// Add new ad types to the end of this list before *_MAX to avoid messing up existing projects.
+	enum AdType {
+		AD_TYPE_BANNER,
+		AD_TYPE_INTERSTITIAL,
+		AD_TYPE_REWARDED,
+		AD_TYPE_MAX,
+	};
+
+	enum BannerAdSize {
+		BANNER_SIZE_SMALL,
```

Comment:
TODO: Figure out how to deal with plugin-specific size formats, e.g. [Admob adaptive banners](https://developers.google.com/admob/android/banner/anchored-adaptive).